tests/framework/db/CommandTest.php

```diff
@@ -321,6 +321,51 @@ public function testInsert()
         ], $record);
     }
 
+    public function testInsertSelect()
+    {
+        $db = $this->getConnection();
+        $db->createCommand('DELETE FROM {{customer}};')->execute();
+
+        $command = $db->createCommand();
+        $command->insert(
+            '{{customer}}',
+            [
+                'email' => 't1@example.com',
+                'name' => 'test',
+                'address' => 'test address',
+            ]
+        )->execute();
+
+	$query = new \yii\db\Query();
+	$query->select([
+                '{{customer}}.email as name',
+                'name as email',
+                'address',
```

Comment:
could you add the alias syntax `'alias' => 'column'`. Seems this is not supported by your implementation right now.

> When the columns are specified as an array, you may also use array keys as the column aliases (if a column
>  does not need alias, do not use a string key).
> http://www.yiiframework.com/doc-2.0/yii-db-query.html#select%28%29-detail

sorry, have not noticed this earlier.